airflow/models/xcom_arg.py

```diff
@@ -0,0 +1,116 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, Dict, List, Optional, Union, Type
+
+from airflow.configuration import conf
+from airflow.models.baseoperator import BaseOperator
+from airflow.models.xcom import XCOM_RETURN_KEY
+from airflow.utils.module_loading import import_string
+
+
+class BaseXComArg:
+    """
+    Class that represents a XCom push from a previous operator.
+    Defaults to "return_value" as only key.
+    """
+
+    def __init__(self, operator: BaseOperator, keys: Optional[List[str]] = None):
+        self._operator = operator
+        self._keys = keys or [XCOM_RETURN_KEY]
```

Comment:
Why keys is a list? I would argue keeping 1 key per instance is better. Having multiple XCOM values associated to a single instance may be confusing.

Curious to hear motivations on this.

Reply:
I think it's mostly related to situations like this
```python
@task
def my_task():
    return 1, 2, 3


with DAG():
    a, b, c = my_task()
```

In this case we can have 3 XCom values or allow to iterate. @jonathanshir

Reply:
I still don't see how having a list of keys in the XComArg would help. XComArg should always be a reference to a single XCom value for clarity/transparency. If we start to bake in more stuff it could become messy imo. 

For the exposed situation, I see a solution to ask the user to return a Dict[str, Any] and parse it such that then the user can do this:

```python
@task(split_xcom_dict=True)
def my_task():
    return {'v1':1, 'v2': 2,'v3': 3}


with DAG():
    d = my_task()
    second_task(d['v1'])
    third_task(d['v2'])
```
I think this is a bit cleaner. Also allows for dependencies to be set correctly as needed and potentially to track data lineage more accurately. Thoughts?

Reply:
I see your point and I know there is some additional magic required to handle multiple outputs.

On the other hand, I would like to allow users to "do whatever they want" and remove the necessity to remember that the their tasks have to have the special return type. Imho the functional operators should "just work" and users should be able to do everything in a pythonic way.

Reply:
> XComArg should always be a reference to a single XCom value for clarity/transparency. If we start to bake in more stuff it could become messy imo.

But I agree with this. If we can we should return XComArg for each return value.

Reply:
> allow users to "do whatever they want" 

The goal is that, but we also need to keep code clean and avoid making it too complex, otherwise it will be difficult to maintain. There's a tradeoff here. I would start without handling multiple outputs (for now just serialize dict and tuples as such in a single xcom value. Let's create a separate issue to handle this discussion discuss the implementation in a separatet PR. Thoughts?

Reply:
> > allow users to "do whatever they want"
> 
> The goal is that, but we also need to keep code clean and avoid making it too complex, otherwise it will be difficult to maintain. There's a tradeoff here. I would start without handling multiple outputs (for now just serialize dict and tuples as such in a single xcom value. Let's create a separate issue to handle this discussion discuss the implementation in a separatet PR. Thoughts?

OK, let's remove _runtime_sub_keys implementation. let's wait till the implementation of multiple outputs for the function is done, and see if it can be done without it.

Reply:
Removed for now :)